ui/components/component-library/button-primary/button-primary.test.js

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable jest/require-top-level-describe */
+import { render } from '@testing-library/react';
+import React from 'react';
+import { ButtonPrimary } from './button-primary';
+import { BUTTON_PRIMARY_SIZES } from './button-primary.constants';
+
+describe('ButtonPrimary', () => {
+  it('should render button element correctly', () => {
+    const { getByText, getByTestId, container } = render(
+      <ButtonPrimary data-testid="button-primary">
+        Button Primary
+      </ButtonPrimary>,
+    );
+    expect(getByText('Button Primary')).toBeDefined();
+    expect(container.querySelector('button')).toBeDefined();
+    expect(getByTestId('button-primary')).toHaveClass('mm-button');
+  });
+
+  it('should render anchor element correctly', () => {
+    const { getByTestId, container } = render(
+      <ButtonPrimary as="a" data-testid="button-primary" />,
+    );
+    expect(getByTestId('button-primary')).toBeDefined();
+    expect(container.querySelector('button')).toBeDefined();
+    expect(getByTestId('button-primary')).toHaveClass('mm-button');
+  });
```

Comment:
How does this test keep passing lol there shouldn't be a `button` in the DOM if we are changing the root element to an `a`. So confused 😅 🤷